### ISSUE TITLE
Native support for packaging multiple layers of same type into a single file

### ISSUE THREAD

**Issue by Hinlavic** [qgis/QGIS#42525]
**Feature description.**
I want to package several layers of same geometry type into a KML file. 
I try to import the layers into a existing  KML file one by one, but it's incredibly slow.

**Additional context**

**Comment by gioman** [comment]
@Hinlavic like "merge vector layer" in the Processing toolbox?

**Comment by jgrocha** [comment]
This is already supported. You can export as many layers as you want to the same kml file. You have to export one by one to the same kml file.

If you want to export many layers to kml at once, you can develop a simple processing algorithm based on the existent 'Package layers' tool, for example. That would solve your use case. 

It could be your contribution to the project. Are you able to accept the challenge?

![package layers tool](https://user-images.githubusercontent.com/163681/112759531-502ac980-8feb-11eb-8fc7-ae0e12c221cf.png)

**Comment by Hinlavic** [comment]
> This is already supported. You can export as many layers as you want to the same kml file. You have to export one by one to the same kml file.
> 
> If you want to export many layers to kml at once, you can develop a simple processing algorithm based on the existent 'Package layers' tool, for example. That would solve your use case.
> 
> It could be your contribution to the project. Are you able to accept the challenge?
> 
> ![package layers tool](https://user-images.githubusercontent.com/163681/112759531-502ac980-8feb-11eb-8fc7-ae0e12c221cf.png)

I would like to, but I am not familiar with coding.

**Comment by Hinlavic** [comment]
> @Hinlavic like "merge vector layer" in the Processing toolbox?

No, I mean the similar function like the "Package Layers" tool that the QGIS already have; However, it only support package into GeoPackage.

**Comment by gioman** [comment]
> No, I mean the similar function like the "Package Layers" tool that the QGIS already have; However, it only supports packaging into GeoPackage. Is that possible getting it work with other popular file format？

@Hinlavic I think you are making some confusion: KML is not a multi layered datasource (as far as I know, but I may be wrong), so the "package layers" tool won't help you (as its objective is to put several separated layers into one single GPKG). On the other hand if you want to merge multiple layers into a single KML file/layer then "merge vector layer" works ok.

**Comment by jgrocha** [comment]
@gioman KML is multi layer. It is supported by QGIS. You can export as many layers as you want to the same KML file. All layers (and styles) will be included in the same KML file.

This feature request makes sense. We can improve the "Package layers" tool to select the target provider type, for all providers supporting multi layers, like Geopackage, PostgreSQL, KML, etc.

**Comment by gioman** [comment]
> @gioman KML is multi layer.

Sorry my bad, I wasn't able to create a multilayer KMl with ogr2ogr so I assumed it was not a thing.

**Comment by alexbruy** [comment]
Closing in favour of #47054.
